Add mqtt broker evaluation (#20)
EURODEO/e-soh-poc-report: source/mqtt_broker_evaluation.adoc

Technologies investigated:

* Mosquitto
* RabbitMQ

> * nat jetstream + mqtt enabled

Reviewer: ```suggestion
* nats jetstream + mqtt enabled
```

* HiveMQ
* EMQX
* VerneMQ

> Mosqitto, HiveMQ and EMQX was disregarded due to lack of needed features or such features beeing behind a paywall.

Reviewer: Can you mention the needed features?
```suggestion
Mosquitto, HiveMQ and EMQX was disregarded due to lack of needed features or such features beeing behind a paywall.
```

> Mosquitto, HiveMQ and EMQX was disregarded due to lack of needed features or such features beeing behind a paywall.

Reviewer: It would still be nice to have these features mentioned:
```suggestion
Requirements:

* req 1
* req 2
* ...

Mosquitto, HiveMQ and EMQX was disregarded due to lack of .. or .. being behind a paywall.
```

> == RabbitMQ

Reviewer: ```suggestion
==== RabbitMQ
```

* TLS support: Yes
* Message persistence support: Yes
* Authentication: Yes
* Clustering support: Yes
* MQTT protocol support: v3.1.1 (as a plugin). RabbitMQ 3.13, the next release series, is expected to be released end of 2023 and will support MQTT protocol version 5.0.
* Free/Open source: Yes

* Pros:
** Wide adoption
** Supported by VMware
** Supports all needed features
** Also has support for other message protocols
* Cons:
** Does not support mqtt v5 as of latest release

> == VerneMQ

Reviewer: ```suggestion
==== VerneMQ
```

* TLS support: Yes
* Message persistence support: No
* Authentication: Yes
* Clustering support: Yes
* MQTT protocol support: v3.1.1  and v5
* Free/Open source: Yes

* Pros:
** Supports all needed features
** Only supports mqtt, less complex
* Cons:

> ** No possiblity for message persistance

Reviewer: ```suggestion
** No possiblity for message persistence, but this is ok according to the design decisions in E-SOH. The API endpoints will provide possibility to retrieve data 24 hours back in time.
```

> == Need resolving before descion

Reviewer: ```suggestion
==== Need resolving before descion
```

> * Do we need assurenace of mqtt v5 before descion, RabbitMQ seems to be lanching v5 before end of year.

Reviewer: ```suggestion
* Do we need assurance of mqtt v5 before deciding? RabbitMQ seems to be launching v5 before the end of the year.
```

> * Are METno going to use the broker, do METno need persistance

Reviewer: ```suggestion
* Is MET Norway going to use the broker, and does MET Norway need persistence?
```

> * Clients must have a way to retrive old messages without persitance

Reviewer: ```suggestion
* Clients must have a way to retrieve old messages if a solution without persistence is selected, but the EDR API should be able to handle this.
```
